### ISSUE TITLE
Implement forgot password for Venia (PWA)

### ISSUE THREAD

**Issue by ericerway** [magento-research/pwa-studio#363]
## This issue is for the following packages:

[X] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-module`
[ ] `pwa-devdocs`

## Description

As a shopper who forgot my password, I want to reset my password for Venia so that I can access previous orders and purchase using my information on file.

Implement a "forgot password" workflow, starting from the sign-in page, which displays a user ID form and a submit button, and then informs the user that if the account exists, they will receive an email to reset their password.

**The backend password reset logic and the sending of the email in this story are out of scope. The Venia app should call the REST API endpoint for resetting the password, and Magento performs the reset and sends the email. There will be a later story for how PWA implementations might also customize emails.**

https://magento.invisionapp.com/share/WQN5F7BYBPG#/318790635_My_Account_-_Sign_In

![image](https://user-images.githubusercontent.com/2686596/47813333-c3e02380-dd18-11e8-8fbd-1f65e8cf676e.png)

https://magento.invisionapp.com/share/WQN5F7BYBPG#/324119948_My_Account_-_Forgot_Password

![image](https://user-images.githubusercontent.com/2686596/46972804-11a42d00-d085-11e8-9fde-b82f6d0802cd.png)

![image](https://user-images.githubusercontent.com/2686596/47037369-5fcc3580-d144-11e8-8f4f-6138c045f28e.png)

#### Assumptions
- Completed My Account (Account) experience Venia OR My Account button is a dummy button today
- Magento backend performs the actual email creation and sending.

## Tasks
[ ] Implement Forgot Password screen with instructions, email and Submit
[ ] Hook up action creators, reducers, and code which calls relevant REST API endpoints
[ ] Testing and documentation (https://github.com/magento-research/pwa-studio/wiki/Documentation-Guidelines)

**Comment by ericerway** [comment]
Kamino cloned this issue to [magento-research/pwa-studio](https://github.com/magento-research/pwa-studio/issues/390)

**Comment by ericerway** [comment]
Kamino cloned this issue to [magento-research/pwa-studio](https://github.com/magento-research/pwa-studio/issues/467)

**Comment by Starotitorov** [comment]
@soumya-ashok, @ericerway, @awilcoxa, @zetlen What should be a behavior for "Continue shopping" button, according to mockups we need to close drawer. There is no problem to implement this. But maybe it is better to navigate user to sign in form and change text of the button? Because after changing password user will try to login once again

**Comment by soumya-ashok** [comment]
@Starotitorov Once the user enters their email, they will actually need to go to their email, and click a link - that link may or may not open in the same browser depending on if they're on an iOS or Android device. They will need to choose a new password there and be prompted to sign in.

This continue shopping is to keep them on the site in the mean time and will take them back to the screen they were on before they attempted to sign in.

@jimbo does that sound reasonable?

**Comment by zetlen** [comment]
We first envisioned this scenario taking place entirely in a drawer, @Starotitorov. Conceptually, this would mean that the "Continue Shopping" button simply dismisses the drawer, just like the close button.  In terms of app flow, this would the drawers to be routed application states, as we discussed in October, with metadata describing the page that underlaid the drawer when it opened. We can implement that later; since that's the ultimate goal and we presently don't have that routed drawer, the "Continue Shopping" button can simply be a link to the homepage.
